.github/workflows/backend-image.yml

```diff
@@ -0,0 +1,65 @@
+name: build-backend
+
+on:
+  push:
+    branches:
+      - master
+      - development
+    tags:
+      - "v*"
+    paths:
+      - "apps/backend/**"
+      - "!apps/backend/readme.md"
+      - "libs/backend-types/**"
+      - "!libs/backend-types/readme.md"
+  pull_request:
+    branches:
+      - development
+    paths:
+      - "apps/backend/**"
+      - "!apps/backend/readme.md"
+      - "libs/backend-types/**"
+      - "!libs/backend-types/readme.md"
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}-backend
+
+jobs:
+  build-container:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          labels: |
+            org.opencontainers.image.title=oko-wallet-backend
+            org.opencontainers.image.description="Backend for OKO wallet app."
+          tags: |
+            type=edge,branch=main
+            type=edge,branch=development,suffix=-dev
+            type=ref,event=tag
+            type=ref,event=pr
+            type=sha
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@v3
+        with:
+          file: ./backend.Dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
```

Comment:
Consider passing Github Actions caching provider to build action.
i.e.
```yaml
cache-from: type=gha
ache-to: type=gha,mode=max
```

More information at: https://docs.docker.com/build/cache/backends/gha/

Reply:
Added)